eras/shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -25,10 +24,15 @@ import qualified Cardano.Ledger.Shelley.LedgerState as LS
   ( returnRedeemAddrsToReserves,
   )
 import Cardano.Ledger.Shelley.TxWits (decodeWits)
+import Cardano.Ledger.ShelleyMA (MaryOrAllegra (Allegra), ShelleyMAEra)
 import Control.Monad.Except (throwError)
 import Data.Coerce (coerce)
 import qualified Data.Map.Strict as Map
 
+--------------------------------------------------------------------------------
+
+type AllegraEra = ShelleyMAEra 'Allegra
+
```

Comment:
Because of the 19 uses of AllegraEra in this file, and the fact that one could not import it due to circular imports. Like for Mary, I moved the type synonym of AllegraEra to   Cardano.Ledger.ShelleyMA.Era, and import it from there, and have Cardano.Ledger.Allegra reexport it.